storage/id_alloc.go

```diff
@@ -77,9 +89,14 @@ func (ia *IDAllocator) Allocate() int64 {
 // to occur before IDs run out to hide Increment latency.
 func (ia *IDAllocator) allocateBlock(incr int64) {
 	ir := &proto.IncrementResponse{}
-	if err := ia.db.Call(proto.Increment, proto.IncrementArgs(ia.idKey, incr), ir); err != nil {
-		log.Fatalf("unable to allocate %d %q IDs: %v", ia.blockSize, ia.idKey, err)
-	}
+	retryOpts := IDAllocationRetryOpts
+	util.RetryWithBackoff(retryOpts, func() (util.RetryStatus, error) {
+		if err := ia.db.Call(proto.Increment, proto.IncrementArgs(ia.idKey, incr), ir); err != nil {
+			return util.RetryContinue, nil
```

Comment:
This should really generate warnings.

    log.Warningf("unable to allocate %d ids from %q: %s", incr, ia.idKey, err)